rtcpeerconnection.go

```diff
@@ -829,6 +834,20 @@ func (pc *RTCPeerConnection) dataChannelEventHandler(e network.DataChannelEvent)
 			fmt.Printf("No datachannel found for streamIdentifier %d \n", e.StreamIdentifier())
 
 		}
+	case *network.DataChannelOpen:
+		for _, dc := range pc.dataChannels {
+			dc.Lock()
+			err := dc.sendOpenChannelMessage()
+			if err != nil {
+				fmt.Println("failed to send openchannel", err)
+			}
+			dc.ReadyState = RTCDataChannelStateOpen
+			dc.Unlock()
+
+			if dc.OnOpen != nil {
```

Comment:
I would guard this with a Lock (Or RLock) the race detector might throw warnings otherwise.